src/field.rs

```diff
@@ -109,27 +109,36 @@ impl Debug for FieldElement {
 impl Index<usize> for FieldElement {
     type Output = Limb;
 
+    #[inline]
     fn index(&self, _index: usize) -> &Limb {
         &(self.0[_index])
     }
 }
 
 impl IndexMut<usize> for FieldElement {
+    #[inline]
     fn index_mut(&mut self, _index: usize) -> &mut Limb {
         &mut(self.0[_index])
     }
 }
 
 impl<'b> AddAssign<&'b FieldElement> for FieldElement {
+    #[inline]
     fn add_assign(&mut self, _rhs: &'b FieldElement) { // fsum()
         for i in 0..self.0.len() {
             self[i] += _rhs[i];
         }
     }
 }
 
+add_impl!(FieldElement, FieldElement, FieldElement);
+sub_impl!(FieldElement, FieldElement, FieldElement);
+mul_impl!(FieldElement, FieldElement, FieldElement);
+neg_impl!(FieldElement, FieldElement);
```

Comment:
This syntax is a bit confusing -- maybe it would be better if the macro invocation looked like
```
mul_impl!( LHS = FieldElement, RHS = FieldElement, Output = FieldElement )
neg_impl!( Self = FieldElement, Output = FieldElement )
```
or something similar, so that the roles of the parameters were clear just by looking?